tiledb/cloud/udf.py

```diff
@@ -98,8 +114,288 @@ def exec(
     """
     return exec_async(
         func=func,
+        name=name,
         arguments=arguments,
         namespace=namespace,
         image_name=image_name,
         http_compressor=http_compressor,
     ).get()
+
+
+def register_udf(func, name, namespace=None, image_name=None, type=None):
+    """
+
+  :param func: function to register
+  :param name: name of udf to register
+  :param namespace: namespace to register in
+  :param image_name: optional image name
+  :param type: type of udf, generic or single_array
+  :return:
+  """
+
+    try:
+        api_instance = client.client.udf_api
+
+        # If the namespace is not set, we will default to the user's namespace
+        if namespace is None:
+            # Fetch the client profile for username if it is not already cached
+            if config.user is None:
+                config.user = client.user_profile()
+
+            namespace = config.user.username
+
+        if not callable(func):
+            raise TypeError("First argument to `exec` must be callable!")
+
+        pickledUDF = cloudpickle.dumps(func, protocol=tiledb_cloud_protocol)
+        pickledUDF = base64.b64encode(pickledUDF).decode("ascii")
+
+        udf_model = rest_api.models.UDFRegistration(
+            name=name,
+            language=rest_api.models.UDFLanguage.PYTHON,
+            version="{}.{}.{}".format(
+                sys.version_info.major, sys.version_info.minor, sys.version_info.micro,
+            ),
+            image_name=image_name,
```

Comment:
If `image_name` is None does it just use the default for `UDFLanguage.PYTHON`?

Reply:
Yes, server side None/empty string is converted to the string "default" which we then parse are default for the language and language version.